### ISSUE TITLE
Itération 3 - Wysiwyg - Import de documents

### ISSUE THREAD

**Issue by Nicoloye** [Jardin-des-Sciences/website#29]
Cette partie me paraît très compliquée à mettre en place telle que je la comprends.
Le wysiwyg proposera une fonctionnalité de nettoyage du copier/coller depuis un fichier word ou open office (avec les limites connues de ces filtres), est-ce suffisant ?

**Comment by Issenmann** [comment]
L'idée, dans le cas de textes longs, était effectivement de pouvoir importer un document rédigé dans word ou open office plutôt que de le saisir directement dans le wysiwyg de la plateforme, en évitant les soucis liés au copier/coller (retours à la ligne abusifs etc) donc si le nettoyage évoqué intervient à ce niveau, c'est bon.

**Comment by bzg** [comment]
Oui, l'idéal serait un import léger, permettant de remplacer le copier-coller+nettoyage pour les documents longs.

**Comment by Nicoloye** [comment]
J'ai fait quelques recherches sur le sujet et je suis en mesure de dire qu'il n'existe aujourd'hui aucun module proposant la possibilité de réaliser un import word/open office sur un champ de contenu Drupal.
J'ai également regardé du côté des éditeurs wysiwyg comme TinyMCE, le mieux que l'on retrouve est cette fameuse fonctionnalité de nettoyage d'un copier/coller avec les limites connues en terme d'efficacité.

Il existe des librairies tierces sur lesquelles nous pourrions nous baser mais elles n'offrent jamais de couverture complète de compatibilité quant au contenu importé (il y aura toujours du nettoyage à faire sur le contenu). Je reporte tout de même la charge d'un tel développement mais je le déconseille au vu du gain potentiel.

**Comment by Issenmann** [comment]
c'est noté, on fait le point sur le coût et sinon nous pourrons peut-être nous contenter du nettoyage standard.

**Comment by bzg** [comment]
@Nicoloye est-ce que vous avez pu tester https://www.drupal.org/project/odfimport ?

Vu que ça n'importe pas les styles du document .odt, cela peut-être servir à importer des documents .odt en les nettoyant?  Je serais intéressé de savoir ce que ça donne à l'usage - si le plugin est resté compatible avec des versions plus récents de Drupal, évidemment.

**Comment by Nicoloye** [comment]
@bzg Je l'ai utilisé il y a très longtemps. De mémoire elle reposait sur des extensions PHP assez peu documentée et les résultats étaient variables selon la complexité des documents.
Nous pouvons tenter un portage du module vers Drupal 7, mais il faudra encore trouver une solution word et s'assurer de la qualité finale.
En tout cas oui ce peut être une base de départ.

**Comment by bzg** [comment]
OK.  De toutes façons, je n'ai jamais été très "fan" de cette idée d'import doc(x), odt... je pense que le copier-coller avec nettoyage rendra aussi bien au final - et si on permet les documents complexes, les utilisateurs vont demander la gestion des notes de bas de page, etc :/

**Comment by Issenmann** [comment]
oui, ok pour les remarques de Bastien, l'idée est de faire simple!
